### DISCUSSION TITLE
How to change default values for fabric object?

### DISCUSSION THREAD

**Discussion by hamzawain7** [fabricjs/fabric.js#9924] [Q&A] [answered]
I used to do this in v5

setDefaults() => {
  window.fabric.Object.prototype.set({
    borderColor: ItemBorderColor.STATIC_ITEM,
    borderOpacityWhenMoving: 1,
    borderScaleFactor: ITEM_CONTROL_DIMENSIONS.BORDER_THICKNESS,
    cornerSize: getCornerSelectableSize(),
    padding: ITEM_CONTROL_DIMENSIONS.PMW_CONTROL_PADDING,
    transparentCorners: false,
  });
  window.fabric.Object.prototype.controls.ml.setVisibility(false);
  window.fabric.Object.prototype.controls.mt.setVisibility(false);
  window.fabric.Object.prototype.controls.mr.setVisibility(false);
  window.fabric.Object.prototype.controls.mb.setVisibility(false);
  }
  
  How do I do this in Fabric V6? I see you are using ownDefaults and getDefaults in the FabricV6 codebase, but how can I nicely override these in my codebase without changing values in the fabricjs code?

**Comment by asturur** [comment, marked as answer]
You can import FabricObject and change FabricObject.ownDefaults with new values for static properties.
For controls you can overide a method called `FabricObject.createControls()` that is a static method and that returns a controls object:

```ts
  /**
   * Creates the default control object.
   * If you prefer to have on instance of controls shared among all objects
   * make this function return an empty object and add controls to the ownDefaults
   * @param {Object} [options] Options object
   */
  static createControls(): { controls: Record<string, Control> } {
    return { controls: createObjectDefaultControls() };
  }
```

Now this method is meant to be called every time and every object has different instances of controls.

If you don't like that, you can have this method return an empty object and add your controls in ownDefaults.
That will give you a situation that is more similar to 5.3.0

You have to be aware of what are you doing to avoid unexpected behavior

**Reply by jjongsu** [reply]
I used ver5.3.0.
I want to change version 6.0.2.

In my case,

at ver5.3.0

fabric.Object.prototype.hasControls = false;
fabric.Object.prototype.lockMovementX = true;
fabric.Object.prototype.lockMovementY = true;
fabric.Object.prototype.hasBorders = false;
fabric.Object.prototype.objectCaching = false;
fabric.Object.prototype.selectable = false;

and change at ver6.0.2

fabric.FabricObject.prototype.hasControls = false;
fabric.FabricObject.prototype.lockMovementX = true;
fabric.FabricObject.prototype.lockMovementY = true;
fabric.FabricObject.prototype.hasBorders = false;
fabric.FabricObject.prototype.objectCaching = false;
fabric.FabricObject.prototype.selectable = false;

But it's not working like it used to.

How can I resolve this issue?

**Reply by asturur** [reply]
https://fabricjs.github.io/docs/configuring-controls/

**Reply by jjongsu** [reply]
@asturur Thank you for your kind support!
